packages/core/botpress/src/dialog/processors.js

```diff
@@ -20,11 +20,11 @@ module.exports = {
         })
       }
 
-      if (/^{.+}$/.test(rendered)) {
+      if (rendered && rendered[0] === '{' && rendered[rendered.length - 1] === '}') {
         // Check if it's JSON
-        Object.assign(additionalData, JSON.parse(rendered))
-      } else if (!_.isEmpty(rendered)) {
-        Object.assign(additionalData, { text: rendered })
+        additionalData = { ...additionalData, ...JSON.parse(rendered) }
```

Comment:
Just curious, is allocating a new object + destructuring faster than Object.assign?

Reply:
At least at some point I've noticed a small perf gain when I removed calls to Object.assign. Which is strange actually because in a compatible env this is transpiled to a call to Object.assign anyways. But I also find this form more readable / explicit.